config/config_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright © 2022 Meroxa, Inc & Yalantis.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestParse(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		cfg map[string]string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    Config
+		wantErr bool
+	}{
+		{
+			name: "success",
+			args: args{
+				cfg: map[string]string{
+					KeyConnection: "sqlserver://sa:password@0.0.0.0?database=mydb&connection+timeout=30",
+					KeyTable:      "CLIENTS",
+				},
+			},
+			want: Config{
+				Connection: "sqlserver://sa:password@0.0.0.0?database=mydb&connection+timeout=30",
+				Table:      "CLIENTS",
+			},
+			wantErr: false,
+		},
+		{
+			name: "fail, missed connection",
+			args: args{
+				cfg: map[string]string{
+					KeyConnection: "",
+					KeyTable:      "CLIENTS",
+				},
+			},
+			want:    Config{},
+			wantErr: true,
+		},
+		{
+			name: "fail, missed table",
+			args: args{
+				cfg: map[string]string{
+					KeyConnection: "sqlserver://sa:password@0.0.0.0?database=mydb&connection+timeout=30",
```

Comment:
maybe we should have this keyConnection string as a constant for this file

Reply:
Good idea !
 done